framework/src/eventloop/eventloop_timer.c

```diff
@@ -0,0 +1,172 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
+#include <libtuv/uv.h>
+#include <eventloop/eventloop.h>
+
+#include "eventloop_internal.h"
+
+typedef struct el_timer_s {
+	timeout_callback func;
+	void *cb_data;
+	event_timer *handle;
+	struct el_timer_s *next;
+} el_timer_t;
+
+el_timer_t *g_timers = NULL;
+
+static void add_timer_list(el_timer_t *timer)
+{
+	el_timer_t *iter;
+
+	if (g_timers == NULL) {
+		g_timers = timer;
+		return OK;
+	}
+
+	iter = g_timers;
+
+	/* Add a new timer to the end of timer list */
+	while (iter->next) {
+		iter = iter->next;
+	}
+	iter->next = timer;
+}
+
+static void remove_timer_list(event_timer *handle)
+{
+	el_timer_t *iter;
+	el_timer_t *next_iter;
+
+	if (g_timers == NULL) {
```

Comment:
the handle value also needs a null check.

Reply:
That is checked before calling this function, line 161.

Reply:
yes, you right.!

Reply:
Every function should check whether parameters are null even though some callers already check this,
because some other callers might miss checking.
It's also a readability matter.